rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/shared_functions.hpp

```
@@ -229,6 +229,13 @@ wait(const char * implementation_identifier,
           }
         }
         if (fixed) {
+          // Reset the fixed guard conditions to avoid being woken up
+          // immediately next time.
+          DDSGuardCondition * guard = static_cast<DDSGuardCondition *>((*attached_conditions)[i]);
+          retcode = guard->set_trigger_value(DDS_BOOLEAN_FALSE);
+          if (retcode != DDS_RETCODE_OK) {
+            RMW_SET_ERROR_MSG("failed to set trigger value");
```

You need to return after this I think. Otherwise the error would be ignored.